survival_sponsorship_compassion/models/recurring_contract_line.py

```diff
@@ -0,0 +1,36 @@
+##############################################################################
+#
+#    Copyright (C) 2023 Compassion CH (http://www.compassion.ch)
+#    Releasing children from poverty in Jesus' name
+#    @author: Simon Gonzalez <sgonzalez@ikmail.com>
+#
+#    The licence is in the file __manifest__.py
+#
+##############################################################################
+
+from odoo import fields, models, api
+from odoo.exceptions import UserError
+
+
+class SponsorshipContract(models.Model):
+    _inherit = ["recurring.contract.line"]
+    _name = "recurring.contract.line"
+
+    contract_type = fields.Selection(
+        selection_add=[('CSP', 'Survival Sponsorship')],
+        related="contract_id.type",
+        readonly=True,
+    )
```

Comment:
I don't think you need to  inherit the related field for adding the selection value. Since the original field is inherited, this one should automatically follow in my opinion.